### PR TITLE
TS-4688 handle DNS compression labels in SRV responses

### PR DESCRIPTION
**Pull request by jacksontj**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/451/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/346/ for details.

File: iocore/dns/DNS.cc [hunk 1]
```diff
@@ -1514,24 +1514,24 @@ dns_process(DNSHandler *handler, HostEnt *buf, int len)
...
-        srv->port        = ink_get16(srv_off + SRV_PORT);
-        srv->priority    = ink_get16(srv_off + SRV_COST);
-        srv->weight      = ink_get16(srv_off + SRV_WEIGHT);
-        srv->host_len    = r;
-        srv->host[r - 1] = '\0';
-        srv->key         = makeHostHash(srv->host);
+        srv->port     = ink_get16(srv_off + SRV_PORT);
+        srv->priority = ink_get16(srv_off + SRV_COST);
+        srv->weight   = ink_get16(srv_off + SRV_WEIGHT);
+        srv->host_len = ::strlen(srv->host) + 1;
```
Comment (SolidWallOfCode) [inline iocore/dns/DNS.cc]: Why the `+1`? I wouldn't think `host_len` should include the terminating null. Did it before when it was set to the return value of `ink_ns_name_ntop`?

**Comment by SolidWallOfCode** [comment]
The purpose of this is to use domain name expansion on the name server FQDN?

**Comment by jacksontj** [comment]
Yes, and we actually already do that for seemingly all the other types-- but not for SRV :/

File: iocore/dns/DNS.cc (hunk 1 above)
Comment (SolidWallOfCode) [inline iocore/dns/DNS.cc]: `n` doesn't have the value you want here?

Comment (jacksontj) [inline iocore/dns/DNS.cc]: Sadly no, `n` is the length of the compressed string, not the expanded string-- meaning I have to calculate the len of the expanded string.

**Pull request merged**